src/applications/mhv/medical-records/components/RecordList/RecordList.jsx

```diff
@@ -1,35 +1,42 @@
 import React, { useEffect, useState, useRef } from 'react';
-import { useDispatch, useSelector } from 'react-redux';
 import PropTypes from 'prop-types';
 import { chunk } from 'lodash';
 import { focusElement } from '@department-of-veterans-affairs/platform-utilities/ui';
 import { VaPagination } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { useHistory } from 'react-router-dom/cjs/react-router-dom.min';
 import RecordListItem from './RecordListItem';
-import { setPagination } from '../../actions/pagination';
-
+import { getParamValue } from '../../util/helpers';
 // Arbitrarily set because the VaPagination component has a required prop for this.
 // This value dictates how many pages are displayed in a pagination component
 const MAX_PAGE_LIST_LENGTH = 5;
 const RecordList = props => {
   const { records, type, perPage = 10, hidePagination } = props;
   const totalEntries = records?.length;
-  const paginationPage = useSelector(state =>
-    state.mr.pagination.page.find(key => key.domain === type),
-  );
 
-  const dispatch = useDispatch();
+  const history = useHistory();
+  const { location } = window;
```

Comment:
Can you use `const location = useLocation()` here instead of `window.location`? It might simplify things in your useEffect below. React Router's `location` is designed to keep track of changes to the page location, so you can probably use that instead of the `window.onpopstate` below. Just put `location` in the useEffect's dependency array and it will trigger whenever the location changes.